src/main/scala/workshop3/03 option.md

```diff
@@ -0,0 +1,122 @@
+Option
+======
+
+The `map`, `filter`, `foreach` and `fold` operators which we saw in the last section are not only found in `List`, 
+but also in other data structures. In this section we will cover the `Option` type.
+
+Consider the following function which either returns a `String`, or a `null`, if the input is invalid
+
+```scala
+def makeStringOfLength(n: Int): String = {
+  if (n <= 0) null
+  else {
+    val x = "abcdef"
+    x * (n / x.length) + x.take(n % x.length)
+  }
+}
+```
+
+**Side note**: `x` in the code snippet above is a `String` and it has an operator `*` defined on it, which concatenates `x` 
+a certain number of times to itself. `x * 3` would result in `abcdefabcdefabcdef`. The `take` operator on the other hand 
+takes the first `n` characters in a `String` and discards the others. `x.take(3)` would result in `abc`.
+
+To use this function we always need to check for a potential `null` pointer. Notice, however, that this is not specified 
+in the type of this function, which is `Int => String`. Unless the function is well documented or you have access to 
+the source code, you do not have any clue as to when (or if) it can possibly return a `null`.
+
+```scala
+val s = makeStringOfLength(8)
+if (s != null) {
+  println(s.reverse)
+}
+```
+
+These null checks can make your code less readable and also looks ugly! Besides that, it is easy to forget them, which 
+can easily introduces bugs that are hard to find. To avoid these null checks, Scala introduces the `Option` type. By wrapping 
+a value in an `Option`, you indicate that it can either be a value or a `null`. `Option[String]` (pronounced as '*Option of String*') 
+therefore declares to be either a value of type `String` or a `null`.
+
+The `Option` type is an abstract class that has two children: `Some` and `None`. The former indicates the case where there is a 
+value, whereas the latter signals a `null`. To create an `Option`, you can either use these subclasses directly or use the equivalent 
+`Option.apply` and `Option.empty` that are defined in the `Option`'s companion object. Often the latter approach is recommended 
+over the former!
+
+```scala
+Some(123) == Option(123)
+None      == Option.empty
+```
+
+With this we can refactor the function above to return an `Option[String]` instead.
+
+```scala
+def makeStringOfLength2(n: Int): Option[String] = {
```

Comment:
You are right about the ambiguity here. Didn't notice it before. Would you suggest removing the numbers and calling all iterations the same?